projects/power_distribution/src/publish_data.c

```diff
@@ -19,17 +30,42 @@ StatusCode power_distribution_publish_data_init(PowerDistributionPublishConfig c
   return STATUS_CODE_OK;
 }
 
+static void prv_partially_publish(SoftTimerId timer_id, void *context) {
+  if (s_current_measurements == NULL) {
```

Comment:
This shouldn't be needed right, since there will always be a sized array created in power_distribution_publish_data_publish()? Is there a reason there would be a NULL condition that I am missing?

Reply:
You're right it shouldn't be null normally, but I like to defensively guard against it anyways - e.g. what if somebody passes NULL for `current_measurements` into `power_distribution_publish_data`, or somebody decides to reset the pointer to NULL at the end of TXing everything down the line. My reasoning was mostly that dereferencing a pointer that's null at some point in the program makes me nervous 😰